frontend/js/src/common/brainzplayer/Queue.tsx

```diff
@@ -0,0 +1,148 @@
+    clearQueue,
+    currentListen,
+    ambientQueue,
+  } = props;
+
+  const [queueNextUp, setQueueNextUp] = React.useState<BrainzPlayerQueue>([]);
+
+  const addQueueToPlaylist = () => {
+    const jspfTrackQueue = queue.map((track) => {
+      return listenToJSPFTrack(track);
+    });
+
+    NiceModal.show(CreateOrEditPlaylistModal, {
+      initialTracks: jspfTrackQueue,
+    });
+  };
+
+  React.useEffect(() => {
+    if (currentListen && queue.length > 0) {
+      const currentListenIndex = queue.findIndex(
+        (track) => track.id === currentListen.id
+      );
+
+      if (
+        currentListenIndex === -1 ||
+        currentListenIndex === queue.length - 1
+      ) {
+        setQueueNextUp([]);
+      } else {
+        const nextUp = queue.slice(currentListenIndex + 1);
+        setQueueNextUp(nextUp);
+      }
+    } else {
+      // If there is no current listen track, show all tracks in the queue
+      setQueueNextUp([...queue]);
+    }
+  }, [queue, currentListen]);
+
+  return (
+    <>
+      {currentListen && (
+        <>
+          <div className="queue-headers">
+            <h4>Now Playing:</h4>
+            <div className="queue-buttons">
+              <button
+                className="btn btn-info btn-sm"
+                data-toggle="modal"
+                data-target="#CreateOrEditPlaylistModal"
+                onClick={addQueueToPlaylist}
+                type="button"
+              >
+                <FontAwesomeIcon icon={faSave} fixedWidth /> Save to Playlist
+              </button>
+              <button
+                className="btn btn-info btn-sm"
+                onClick={clearQueue}
+                type="button"
+              >
+                Clear Queue
+              </button>
+            </div>
+          </div>
+          <ListenCard
+            key={`queue-listening-now-${getListenCardKey(currentListen)}}`}
+            className="queue-item-card"
+            listen={currentListen as Listen}
+            showTimestamp={false}
+            showUsername={false}
+          />
+        </>
+      )}
+      <div className="queue-headers">
+        <h4>Next Up:</h4>
+      </div>
+      <div className="queue-list">
+        {queueNextUp.length > 0 ? (
+          <ReactSortable
+            handle=".drag-handle"
+            list={queueNextUp}
+            onEnd={moveQueueItem}
+            setList={() => {}}
+          >
+            {queueNextUp.map((queueItem: BrainzPlayerQueueItem) => {
+              return (
+                <QueueItemCard
+                  key={queueItem.id}
+                  track={queueItem}
+                  removeTrackFromQueue={removeTrackFromQueue}
+                />
+              );
+            })}
+          </ReactSortable>
+        ) : (
+          <div className="lead text-center">
+            <p>Nothing in this queue yet</p>
+          </div>
+        )}
+      </div>
+      <div className="queue-headers">
+        <h4>Suggested for you:</h4>
```

Comment:
I think the wording of this could be workshopped, we do'nt want users to think that these are recommendations we generate on the spot.
Maybe "On this page:" or something along those lines?
I don't have a better idea right now.
<img width="475" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/6179856/096b9546-37b6-478e-b563-0defcd6a0438">


Another option could be to remove the header (but keep the divider, with some padding) and show the ambientQueue with a lower opacity like so:
<img width="506" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/6179856/62b4f46d-9471-4fe8-a804-59b1ffb89a90">

What do you think?

Reply:
`On this page` looks good to me